File: web/src/api/h5Admin.js
import service from '@/utils/request'

// --- H5 User Management ---
export const getH5UserList = (params) => service({ url: '/h5Admin/userList', method: 'get', params })
export const getH5User = (params) => service({ url: '/h5Admin/user', method: 'get', params })
export const updateH5User = (data) => service({ url: '/h5Admin/user', method: 'put', data })
export const deleteH5User = (data) => service({ url: '/h5Admin/user', method: 'delete', data })

// --- H5 Post Management ---
export const getH5PostList = (params) => service({ url: '/h5Admin/postList', method: 'get', params })
export const getH5Post = (params) => service({ url: '/h5Admin/post', method: 'get', params })
export const updateH5Post = (data) => service({ url: '/h5Admin/post', method: 'put', data })
export const deleteH5Post = (data) => service({ url: '/h5Admin/post', method: 'delete', data })

// --- H5 Comment Management ---
export const getH5CommentList = (params) => service({ url: '/h5Admin/commentList', method: 'get', params })
export const deleteH5Comment = (data) => service({ url: '/h5Admin/comment', method: 'delete', data })

// --- H5 Tag Management ---
export const getH5TagList = (params) => service({ url: '/h5Admin/tagList', method: 'get', params })
export const updateH5Tag = (data) => service({ url: '/h5Admin/tag', method: 'put', data })
export const deleteH5Tag = (data) => service({ url: '/h5Admin/tag', method: 'delete', data })

// --- H5 Collection Management ---
export const getH5CollectionList = (params) => service({ url: '/h5Admin/collectionList', method: 'get', params })
export const deleteH5Collection = (data) => service({ url: '/h5Admin/collection', method: 'delete', data })

// --- H5 Following Management ---
export const getH5FollowingList = (params) => service({ url: '/h5Admin/followingList', method: 'get', params })
export const deleteH5Following = (data) => service({ url: '/h5Admin/following', method: 'delete', data })

File: server/router/system/h5_admin.go
package system

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type H5AdminRouter struct{}

func (r *H5AdminRouter) InitH5AdminRouter(Router *gin.RouterGroup) {
	g := Router.Group("h5Admin").Use(middleware.OperationRecord())
	{
		g.GET("userList", h5AdminApi.GetH5UserList)
		g.GET("user", h5AdminApi.GetH5User)
		g.PUT("user", h5AdminApi.UpdateH5User)
		g.DELETE("user", h5AdminApi.DeleteH5User)

		g.GET("postList", h5AdminApi.GetH5PostList)
		g.GET("post", h5AdminApi.GetH5Post)
		g.PUT("post", h5AdminApi.UpdateH5Post)
		g.DELETE("post", h5AdminApi.DeleteH5Post)

		g.GET("commentList", h5AdminApi.GetH5CommentList)
		g.DELETE("comment", h5AdminApi.DeleteH5Comment)

		g.GET("tagList", h5AdminApi.GetH5TagList)
		g.PUT("tag", h5AdminApi.UpdateH5Tag)
		g.DELETE("tag", h5AdminApi.DeleteH5Tag)

		g.GET("collectionList", h5AdminApi.GetH5CollectionList)
		g.DELETE("collection", h5AdminApi.DeleteH5Collection)

		g.GET("followingList", h5AdminApi.GetH5FollowingList)
		g.DELETE("following", h5AdminApi.DeleteH5Following)
	}
}
